app/jobs/get_suggestions_job.rb

```diff
@@ -0,0 +1,38 @@
+class GetSuggestionsJob < ApplicationJob
+  queue_as :default
+
+  def perform(job_status)
+    @job_id = job_status.jobId
+
+    @authenticator = SpotifyAuthenticator.new
```

Comment:
What if the result from `SpotifyAuthenticator` could be used to get the track data directly, instead of passing it around in here?

e.g. something like, vaguely,

```ruby
def perform(job_status)
  if spotify.has_tracks?
    job_status.update(job_attributes_success)
  else
    job_status.update(job_attributes_failure)
 end
end

private

def spotify
  @_spotify ||= SpotifyAuthenticator.authenticate
end

def job_attributes_success
  { ...,
    payload: spotify.track_data
  }
end

def job_attributes_failure
  { ...,
  payload: spotify.failure_response
  }
end
```